### ISSUE TITLE
Display purchase options for all users during onboarding

### ISSUE THREAD

**Issue by markmhendrickson** [blockstack/blockstack-app#2251]
To help with conversion into purchase options, we should display them directly on the home screen after registration.

**Comment by landitus** [comment]
Here are the first concepts for adding the purchasing options on the home screen. UX-wise, it would occupy the slot for the Next steps. Once the account is funded, the Next steps will appear. 

[Figma prototype](https://www.figma.com/proto/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?page-id=23%3A1717&node-id=34%3A9314&viewport=241%2C48%2C0.25&scaling=min-zoom&starting-point-node-id=34%3A9314&show-proto-sidebar=1)

<img width="1440" alt="Home A" src="https://user-images.githubusercontent.com/239215/163231847-0abb43a4-6272-4982-b263-8576832ace4d.png">
<img width="1440" alt="Home B" src="https://user-images.githubusercontent.com/239215/163231857-1b032d3b-83f3-43f4-b98a-2c76117db22f.png">

We discussed these options with the team and we concluded:
- There's very little space for all the information required. Eventually, more options will be available as well.
- It feels disjointed from the rest of the home screen: why not place these options as the Activity empty space?
- It adds too many CTAs/buttons, so there's less focus on the primary actions.

So we decided to explore different options during the onboarding flow, which is more focused and comes right before the home screen.

**Comment by landitus** [comment]
Here's another round of concepts, this time focused on introducing the buying options during the initial onboarding flow. 

These concepts play around with information density: 
- How much information do we provide up-front from each provider? payment options, location restrictions
- Do we show other exchanges that don't have Stacks on-ramp integration as well? Showing them provides a larger set of options to choose from (can be argued to add complexity and also to allow for more users to access). Adding all providers brings the need to differentiate the on-ramp integrated providers, in this case, with a label.

Also they some could be re-used and replace the current Buy screen. Though not a priority, it's something we should take into account.

[Figma prototype](https://www.figma.com/proto/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?page-id=66%3A12649&node-id=66%3A17234&viewport=241%2C48%2C0.5&scaling=scale-down&starting-point-node-id=66%3A17234&show-proto-sidebar=1)

<img width="1440" alt="Buy" src="https://user-images.githubusercontent.com/239215/163235414-b2242aa5-9bd0-4b23-8699-fb4518a3e8ce.png">
<img width="1440" alt="Buy-1" src="https://user-images.githubusercontent.com/239215/163235411-2ccc9d2b-fd50-4f84-af5c-2f0bbf32133e.png">
<img width="1440" alt="Buy-2" src="https://user-images.githubusercontent.com/239215/163235406-d486bbd4-e2ff-47d0-ac90-27de060715ca.png">

**Comment by landitus** [comment]
Based on the team's feedback, we want to start a round of Usability tests with one of the concepts and move from there. We choose the grid with all the options (including non-integrated providers). 

- [Figma prototype](https://www.figma.com/proto/WXQDHao57vMzNo348QYFKz/%232251---Fund-account---Usability-test-%231?page-id=126%3A41117&node-id=139%3A25272&viewport=241%2C48%2C0.38&scaling=min-zoom&starting-point-node-id=139%3A25272)
- [Figma prototype for usability testing](https://www.figma.com/proto/WXQDHao57vMzNo348QYFKz/%232251---Fund-account---Usability-test-%231?page-id=126%3A41117&node-id=139%3A25272&viewport=241%2C48%2C0.38&scaling=min-zoom&starting-point-node-id=139%3A25272&hotspot-hints=0&hide-ui=1) (no hints or figma UI)

<img width="1440" alt="Buy - options" src="https://user-images.githubusercontent.com/239215/163236235-83df4896-4de9-4c93-b7d1-fc92e3c63ebe.png">

**Comment by fbwoolf** [comment]
> 
> * [Figma prototype](https://www.figma.com/proto/WXQDHao57vMzNo348QYFKz/%232251---Fund-account---Usability-test-%231?page-id=126%3A41117&node-id=139%3A25272&viewport=241%2C48%2C0.38&scaling=min-zoom&starting-point-node-id=139%3A25272)
> * [Figma prototype for usability testing](https://www.figma.com/proto/WXQDHao57vMzNo348QYFKz/%232251---Fund-account---Usability-test-%231?page-id=126%3A41117&node-id=139%3A25272&viewport=241%2C48%2C0.38&scaling=min-zoom&starting-point-node-id=139%3A25272&hotspot-hints=0&hide-ui=1) (no hints or figma UI)
> 
I don't understand the difference in the 2 Figma links?

**Comment by fbwoolf** [comment]
@markmhx is this something I can start on?

**Comment by markmhendrickson** [comment]
We're planning to run these through some usability testing before settling on the version for integration. So let's hold off for now until we get some of those results back. It should hopefully not take more than a week or so 🙏

**Comment by landitus** [comment]
> I don't understand the difference in the 2 Figma links?

These have different properties in the URL to toggle the visibility of Figma's UI on top of the prototype. For usability testing, it's better to remove any additional UI that's not part of the design.

**Comment by andresgalante** [comment]
Maybe I am late for the party here, just wanted to comment that I really like these concepts and am looking forward to the user testing results

**Comment by landitus** [comment]
We have completed 2 rounds of usability tests. We've got a total of 10 participants in 2 rounds of 5 each. The first round was really exploratory to get familiar with the UserTesting.com platform, and results are mixed because the test questions were not as fleshed out. Still, we catched various copy improvements throughout the onboarding that were confusing to users. 

The second round incorporated these copy changes and added an instructions modal for non on-ramp exchanges like Coinabase and Binance. We got really positive results. **100% thought it was "Very easy" to complete the flow**.  

[Here's the full prototype](https://www.figma.com/proto/qxFRcKhA56LxJX8FG0qlbq/%232251---Fund-account---Usability-test-%232?page-id=126%3A41117&node-id=126%3A41293&viewport=337%2C48%2C0.03&scaling=min-zoom&starting-point-node-id=126%3A41293) 

**Noticiables changes:**
- Copy: `Add Hiro Wallet to Chrome` changes to: `To buy NFTs you need to add Hiro Wallet to Chrome`
- Copy: `Download Hiro Wallet` changes to: `Download Hiro Wallet extension`
- Copy:`Create Stacks account` changes to `Create new wallet`
- Copy:`Fund your account` changes to `Let’s get some STX  into your wallet`
- Copy:`Choose an exchange to fund your account with STX, the native currency of Stacks` changes to `Choose an exchange to fund your account with Stacks (STX)`
- New screen: Instructions to buy and transfer from a non on-ramp exchange, such as Coinbase. It includes the current wallet account's address with a copy button and links to the exchange. The copy is still fake, but we should re-use it for all exchanges by keeping it pretty generic and high-level.

<img width="1279" alt="CleanShot 2022-04-28 at 16 20 22@2x" src="https://user-images.githubusercontent.com/239215/165774505-ec58c6c8-9204-4021-b793-05c3e1703cc1.png">
<img width="1278" alt="CleanShot 2022-04-28 at 16 20 38@2x" src="https://user-images.githubusercontent.com/239215/165774405-b0803266-4edd-4d78-b428-0ffaf04c4dcf.png">
<img width="1277" alt="CleanShot 2022-04-28 at 16 20 51@2x" src="https://user-images.githubusercontent.com/239215/165774418-5c8f71d1-fb06-43f8-8423-8759e249379c.png">
<img width="1280" alt="CleanShot 2022-04-28 at 16 19 51@2x" src="https://user-images.githubusercontent.com/239215/165774555-f2d722a8-c52e-4278-b5b8-7b1c59ffda1b.png">

For the **Connect** modal, we noticed the main copy was shocking to new users so we opted to add more context. The idea would be to provide a generic copy like the one we have today, maybe rephrase it like: To **{continue}** you need to add the Hiro Wallet to {browser}. And devs could pass a phrase like:  To **buy NFTs** you need to add the Hiro Wallet to {browser}. 

### Findings
- Copy changes were successful as people completed the flow smoothly.
- Tags like fast checkout and 0% fees got noticed. Still, people chose the exchange based on familiarity and previous experience
- The buying process was perceived as very easy. Still, some people (novices) hesitated on why they needed to create a wallet to buy an NFT. The experts didn't think twice about it.
- In the future, we should how to set better expectations of what this flow will take (steps, time, visiting 4 different websites). Maybe adding a stepper or wizard like experience. Specially targeted at users new to crypto. 

### Next steps
- Adjust the designs and spec them out for production
- Define instructions for other exchanges (Coinbase)
- Define telemetry to track the completion of the flow

**Comment by landitus** [comment]
Here's the design spec to begin development. This only covers the onboarding Fund account step. The rest of the flow will be split into separate tasks under the same epic.

**[Figma](https://www.figma.com/file/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?node-id=310%3A54443)**

### General

**Logic**
The Fund account screen is a new step of the wallet onboarding flow. After the user creates a Password, they are prompted to fund their account. User can skip and go to the wallet.

**Resuming from closed tab**
If the dismisses the onboarding (ex. by closing the tab). Once they re-open the wallet they will be taken to this screen to resume the onboarding. If they skip this screen, the onboarding is completed and user’s won’t see this screen again.

**Content**
The list of exchanges is curated and handpicked by Hiro based on each of the exchanges UX and wallet integration and popularity. The order may change over time to always show the best options at the top.

**Responsive**
Experience should be responsive, similar to the onboarding flow. There are no special adaptations to the widget format, since the onboarding will always force open in a new tab

![Image](https://user-images.githubusercontent.com/239215/168642567-4a592694-4ee5-4eab-888c-81280abb159f.png)

---

### Components

![Image](https://user-images.githubusercontent.com/239215/168642642-ae7348af-e0a9-464d-8419-52b359b2fa39.png)

![Image](https://user-images.githubusercontent.com/239215/168642671-11015873-63cc-40a7-a008-fb333b05776e.png)

---

### Demo

Check this [demo](https://www.figma.com/proto/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?page-id=220%3A27927&node-id=310%3A54589&viewport=-924%2C-991%2C0.25&scaling=min-zoom&starting-point-node-id=310%3A54443) or the [prototype](https://www.figma.com/proto/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?page-id=285%3A40583&node-id=285%3A40709&viewport=783%2C409%2C0.03&scaling=scale-down&starting-point-node-id=285%3A40709&show-proto-sidebar=1) for the complete flow 

![Image](https://user-images.githubusercontent.com/239215/168642790-b0726816-caa2-4ee8-95f7-8d751ff82dbb.png)

**Comment by fbwoolf** [comment]
@landitus @markmhx I started working on this but realized there are instructions that onboarding should only happen in full page view ...however, this PR was never merged to implement that behavior: https://github.com/hirosystems/stacks-wallet-web/pull/2266 ...or am I confused here? What happens in the extension on this page?

Should I stop and force the onboarding flow in full page view before starting development on this or are we ok if this step is also shown in the extension?

EDIT: Is this even possible?

I have to refactor the Buy page: https://github.com/hirosystems/stacks-wallet-web/issues/2422 ...so maybe it's best just to start with the page being responsive for both views ...and then implement changes to routing behavior after?

**Comment by markmhendrickson** [comment]
> so maybe it's best just to start with the page being responsive for both views ...and then implement changes to routing behavior after?

That works for me, though curious what @landitus thinks here.

I expect that the vast majority of users will onboard via full screen mode anyway, and despite not yet tackling #2266, since the default post-install behavior is to open a full new tab for the extension. However, it would be best if can format this screen for the narrower modal should some users pass through onboarding in that context anyway (and presuming we can't / aren't forcing the full new tab for this initial implementation).

**Comment by landitus** [comment]
> so maybe it's best just to start with the page being responsive for both views ...and then implement changes to routing behavior after?

Sounds good. But I believe that the soon to be merged Ledger implementation will force the full screen anyways. @kyranjamie we are still on track for this, correct? For context, even if we were reluctant to merge https://github.com/hirosystems/stacks-wallet-web/pull/2266, while working on Ledger we made the decision with @kyranjamie to force the onboarding in a tab. Otherwise, Ledger cannot be configured in the widget.

**Comment by fbwoolf** [comment]
> 
> Sounds good. But I believe that the soon to be merged Ledger implementation will force the full screen anyways. @kyranjamie we are still on track for this, correct?
>
Interesting, so I'll wait for Kyran to respond but maybe I should be branching off the ledger work or holding on this? I started the work branching off dev.

**Comment by kyranjamie** [comment]
I'm going to close https://github.com/hirosystems/stacks-wallet-web/pull/2266, as @landitus said, this will be introduced with Ledger

**Comment by markmhendrickson** [comment]
> maybe I should be branching off the ledger work or holding on this?

Preferably we can keep this work independent from the Ledger work, since it's possible we want to ship it before Ledger goes fully live (post-preview-release).

**Comment by andresgalante** [comment]
@markmhx to decide what to with the text and the FAQ

**Comment by landitus** [comment]
For the time being, each exchange card will display one of the following descriptions, based on US availability. To decide which copy applies to each exchange, we'll need to research each one and check what they support (checking their website/support pages)

<img width="630" alt="CleanShot 2022-06-08 at 08 59 49@2x" src="https://user-images.githubusercontent.com/239215/172623392-0da83165-9b32-41d6-aff1-8c92b928e5d2.png">

- Available in the US and other countries
- Available outside the US and other countries

[Figma](https://www.figma.com/file/mpfjDNidnBwUUoCz8r1Wkx/%232251---Fund-account?node-id=597%3A55697)
